quinn-proto/src/coding.rs

```diff
@@ -1,18 +1,25 @@
+//! Coding related traits.
+
 use std::net::{Ipv4Addr, Ipv6Addr};
 
 use bytes::{Buf, BufMut};
 use thiserror::Error;
 
 use crate::VarInt;
 
+/// Unexpected end of buffer error.
 #[derive(Error, Debug, Copy, Clone, Eq, PartialEq)]
 #[error("unexpected end of buffer")]
 pub struct UnexpectedEnd;
 
+#[doc(hidden)]
 pub type Result<T> = ::std::result::Result<T, UnexpectedEnd>;
 
+/// Trait to handle encoding and decoding from byte buffers.
```

Comment:
This is not a general-purpose serialization interface, and we don't need to restate that it's a trait.

```suggestion
/// Infallible encoding and decoding of QUIC primitives
```